File: src/Services/Ordering/Ordering.Domain/Entities/Order.cs
using Ordering.Domain.Common;

namespace Ordering.Domain.Entities;

public class Order : EntityBase
{
    public required string UserName { get; set; }
    public decimal TotalPrice { get; set; }

    public required ShippingAddress ShippingAddress { get; set; }
    public required Card PaymentCard { get; set; }

    public IEnumerable<Product> Items { get; set; } = new List<Product>();
}

public class ShippingAddress
{
    public required string FullName { get; set; }
    public required string Email { get; set; }
    public required string AddressLine { get; set; }
    public required string Country { get; set; }
    public required string State { get; set; }
    public required string City { get; set; }
    public required int ZipCode { get; set; }
}

public class Card
{
    public required string Name { get; set; }
    public required string Number { get; set; }
    public required DateTime Expiration { get; set; }
    public required string Cvv { get; set; }
}

File: src/WebApps/react-app.client/src/app/models/order.ts
export interface ShippingAddress {
    fullName: string
    email: string
    addressLine: string
    country: string
    state: string
    city: string
    zipCode: number
}

export interface Card {
    name: string
    number: string
    expiration: string
    cvv: string
}

export interface OrderItem {
    name: string
    color: string
    price: number
    quantity: number
    id: string
    createdBy: string
    createdDate: string
    lastModifiedBy: string
    lastModifiedDate: string
}

export interface Order {
    id: string;
    userName: string;
    totalPrice: number;
    shippingAddress: ShippingAddress;
    paymentCard: Card
    items: OrderItem[];
    status: string;
    orderDate: string;
}